R/stitch.rspec.R

```diff
@@ -0,0 +1,155 @@
+#' a single `rspec` object.
+#'
+#' @param rspec1,rspec2 (required) `rspec` objects of differing wavelength ranges
+#' to stitch together by row.
+#' @param overlap_method the method for modifying reflectance values if regions
+#' of the spectra overlap in their wavelength range. Defaults to `mean`.
+#' @param interp logical argument specifying whether reflectance values should be
+#' interpolated between the two sets of spectra if their wavelength ranges
+#' do not overlap. Defaults to `TRUE`.
+#'
+#' @export
+#'
+#' @examples
+#'
+#' # Simulate a UV-VIS and NIR reflectance spectrum whose wavelength regions
+#' # slightly overlap then stitch them together, with the overlapping
+#' # regions being averaged.
+#'
+#' # Simulate specs
+#' reflect1 <- simulate_spec(wl_peak = 550, xlim = c(300, 700))
+#' reflect2 <- simulate_spec(wl_inflect = 1100, xlim = c(650, 1200))
+#'
+#' # Ensure the names of the spectra match
+#' names(reflect1) <- names(reflect2) <- c('wl', 'sample_1')
+#'
+#' # Stitch the spectra together by their wavelength column
+#' full_spec <- stitch(reflect1, reflect2)
+#'
+#' # Plot the resulting spectrum
+#' plot(full_spec)
+#'
+#' # Simulate another set of UV-VIS and NIR spectra. Note two additional complexities,
+#' # both of which are handled without issue. First, the wavelength ranges are
+#' # non-overlapping (with a 100 nm gap). We'll keep the default interp = TRUE argument
+#' # to allow the missing reflectance region to be interpolated. Second, the names of
+#' # the spectra match, but are in a different order in the two rspec objects. This isn't
+#' # an issue, as the function can match up the spectra by name irrespective of their
+#' # ordering
+#'
+#' # Simulate UV-VIS and NIR spectra
+#' reflect_vis <- merge(simulate_spec(wl_peak = 550, xlim = c(300, 700)),
+#'                      simulate_spec(wl_peak = 550, xlim = c(300, 700)))
+#' reflect_nir <- merge(simulate_spec(wl_inflect = 1000, xlim = c(800, 1250)),
+#'                      simulate_spec(wl_inflect = 1100, xlim = c(800, 1250)))
+#'
+#' # Ensure the names of the spectra exist in each, albeit in a different order
+#' names(reflect_vis) <- c('wl', 'sample_1', 'sample_2')
+#' names(reflect_nir) <- c('wl', 'sample_2', 'sample_1')
+#'
+#' # Stitch together by their wavelength column, with missing regions being
+#' # interpolated
+#' reflect_vis_nir <- stitch(reflect_vis, reflect_nir)
+#'
+#' # Plot the resulting spectrum
+#' plot(reflect_vis_nir)
+#'
+#' @author Thomas White \email{thomas.white026@@gmail.com}
+#' @author Hugo Gruson \email{hugo.gruson+R@@normalesup.org}
+#'
+#' @seealso [as.rspec()], [merge.rspec()]
+
+stitch <- function(rspec1, rspec2, overlap_method, interp) {
+  UseMethod("stitch")
+}
+
+#' @rdname stitch
+#'
+#' @export
+stitch.rspec <- function(rspec1, rspec2,
+                                 overlap_method = c("mean", "minimum", "maximum"),
+                                 interp = TRUE) {
```

Comment:
```suggestion
stitch <- function(x1, x2, ...) {
  UseMethod("stitch")
}

#' @rdname stitch
#'
#' @export
stitch.rspec <- function(x1, x2,
                                 overlap_method = c("mean", "minimum", "maximum"),
                                 interp = TRUE) {
```


To leave room for future extensions or other methods.

We may want to re-use this generic for other objects in the future and it's probably safer to offer more flexibility in the argument names. This is the same reason why we use `x` as the first argument in `subset.rspec()`, `plot.rspec()`, etc.

If we don't expect ever re-using this generic for other objects, we could also not leverage S3 and propose a `stitch_rspec()` or `stitchrspec()` function directly. The same way we do it with `explorespec()`, `aggspec()`, etc.